nntrainer/utils/base_properties.h

```diff
@@ -133,28 +160,44 @@ template <typename T> class Property {
    *
    * @return T value
    */
-  operator T &() { return value; }
+  operator T &() { return get(); }
 
   /**
    * @brief cast operator for property
    *
    * @return T value
    */
-  operator const T &() const { return value; }
+  operator const T &() const { return get(); }
 
   /**
    * @brief get the underlying data
    *
    * @return const T& data
    */
-  const T &get() const { return value; }
+  const T &get() const {
+    NNTR_THROW_IF(value == nullptr, std::invalid_argument)
+      << "Cannot get property, property is empty";
+    return *value;
+  }
 
   /**
    * @brief get the underlying data
    *
    * @return T& data
    */
-  T &get() { return value; }
+  T &get() {
+    NNTR_THROW_IF(value == nullptr, std::invalid_argument)
+      << "Cannot get property, property is empty";
+    return *value;
+  }
+
+  /**
+   * @brief check if property is empty
+   *
+   * @return true empty
+   * @return false not empty
```

Comment:
change this to retval

Reply:
is this already done?

Reply:
Yes

Reply:
chronologically,

1. Create this PR
2. Patch return -> retval (merged)
3. this PR patched (today)